lib/base-cmd.js

```diff
@@ -151,6 +151,19 @@ class BaseCommand {
     this.workspaceNames = [...ws.keys()]
     this.workspacePaths = [...ws.values()]
   }
+
+  /** commands like ci use workspace config but not --workspace */
+  async setWorkspaceFlagIndependent () {
```

Comment:
This seems a little overwrought at first glance.  `ci` is a special case, why don't we call `get-workspaces` directly inside there with an empty filter to get this info?